probe-rs/src/bin/probe-rs/cmd/test.rs

```diff
@@ -0,0 +1,617 @@
+use std::cell::RefCell;
+use std::fs::File;
+use std::io::Write;
+use std::ops::Range;
+use std::path::Path;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use anyhow::{anyhow, bail, Context, Result};
+use libtest_mimic::{Failed, Trial};
+use probe_rs::debug::{DebugInfo, DebugRegisters};
+use probe_rs::flashing::{FileDownloadError, Format};
+use probe_rs::{
+    exception_handler_for_core, BreakpointCause, Core, CoreInterface, CoreStatus, Error,
+    HaltReason, Lister, MemoryInterface, SemihostingCommand, Session, VectorCatchCondition,
+};
+use probe_rs_target::MemoryRegion;
+use signal_hook::consts::signal;
+use static_cell::StaticCell;
+use time::UtcOffset;
+
+use crate::util::common_options::{BinaryDownloadOptions, ProbeOptions};
+use crate::util::flash::run_flash_download;
+use crate::util::rtt::{self, RttConfig};
+use crate::FormatOptions;
+
+const RTT_RETRIES: usize = 10;
+
+#[derive(clap::Parser)]
+pub struct Cmd {
+    #[clap(flatten)]
+    pub(crate) probe_options: ProbeOptions,
+
+    #[clap(flatten)]
+    pub(crate) download_options: BinaryDownloadOptions,
+
+    /// Always print the stacktrace on ctrl + c.
+    #[clap(long)]
+    pub(crate) always_print_stacktrace: bool,
+
+    /// Whether to erase the entire chip before downloading
+    #[clap(long)]
+    pub(crate) chip_erase: bool,
+
+    /// Suppress filename and line number information from the rtt log
+    #[clap(long)]
+    pub(crate) no_location: bool,
+
+    #[clap(flatten)]
+    pub(crate) format_options: FormatOptions,
+
+    #[clap(long)]
+    pub(crate) log_format: Option<String>,
+
+    /// Enable reset vector catch if its supported on the target.
+    #[arg(long)]
+    pub catch_reset: bool,
+    /// Enable hardfault vector catch if its supported on the target.
+    #[arg(long)]
+    pub catch_hardfault: bool,
+
+    /// Scan the memory to find the RTT control block
+    #[clap(long)]
+    pub(crate) rtt_scan_memory: bool,
+
+    /// <elf> and the remaining arguments for the test runner (list tests, filter tests etc). Run `probe-rs test -- --help` for more information.
+    #[clap(last(true))]
+    pub(crate) libtest_args: Vec<String>,
+}
+
+static SESSION: StaticCell<Session> = StaticCell::new();
```

Comment:
I would make a bold statement and say those cells are not required. Why do we need them?

Reply:
They are needed because of libtest_mimics `'static` requirement. See here https://github.com/LukasKalbertodt/libtest-mimic/blob/2ed35db51e8aa5fe3b84e946be71f0b00e736a33/src/lib.rs#L109

Reply:
I would not do unsafe calls. But I can have a look. I still feel like it should be possible somehow ^^ You only access it from the same thread I thought which should make it possible.

Reply:
We could refactor libtest_mimic, to allow passing a lifetime, instead of requiring 'static....

Reply:
That would be my preferred approach. But that might not work. I am not sure what it needs the static for. Even in a multithreaded environment you can share stuff without it being static. I guess Arc::new is not const.

Reply:
Ok. I've refactored libtest-mimic to allow passing the lifetime ( https://github.com/LukasKalbertodt/libtest-mimic/pull/34/commits/b6ad4d167c977a5c049859c7b84cd9e363771c5b ), but I'm not sure if this is going to be accepted, since it breaks the public API...

Reply:
Nice! Let's see. But wouldn't a non-static lifetime be more flexible and thus enable the same but more things as with 'static before? :)

Reply:
I think you are referring to an old diff. Its all non static and without cells now.

Reply:
Nono, I mean in libtest-mimic :) I was referring to the question whether it will be accepted or not :)